src/components/Projects/ProjectsFilterAndSort.tsx

```diff
@@ -0,0 +1,153 @@
+
+export default function ProjectsFilterAndSort({
+  includeV1,
+  setIncludeV1,
+  includeV1_1,
+  setIncludeV1_1,
+  includeActive,
+  setIncludeActive,
+  includeArchived,
+  setIncludeArchived,
+  orderBy,
+  setOrderBy,
+}: {
+  includeV1: boolean
+  setIncludeV1: Function
+  includeV1_1: boolean
+  setIncludeV1_1: Function
+  includeActive: boolean
+  setIncludeActive: Function
+  includeArchived: boolean
+  setIncludeArchived: Function
+  orderBy: OrderByOption
+  setOrderBy: any
+}) {
+  const {
+    theme: { colors },
+    isDarkMode,
+  } = useContext(ThemeContext)
+
+  const [activeKey, setActiveKey] = useState<0 | undefined>()
+
+  const filterCollapseStyle: CSSProperties = {
+    border: 'none',
+    marginTop: 0,
+    marginBottom: 0,
+    marginRight: 15,
+  }
+
+  const filterDropdownItemStyles: CSSProperties = {
+    display: 'flex',
+    alignItems: 'center',
+    height: 40,
+  }
+
+  const filterCheckboxItem = (
+    label: string,
+    checked: boolean,
+    onChange: Function,
+    disabled?: boolean,
+  ) => (
+    <div
+      style={{
+        ...filterDropdownItemStyles,
+        color: disabled ? colors.text.tertiary : '',
+        cursor: 'pointer',
+      }}
+      onClick={() => onChange(!checked)}
+    >
+      <Checkbox
+        style={{ marginRight: 10 }}
+        checked={checked}
+        onChange={() => onChange(!checked)}
+        disabled={disabled}
+      />
+      <Trans>{label}</Trans>
+    </div>
+  )
+
+  // If includeArchived is true, we set active back to true
+  // to avoid both being unselected
+  const toggleArchived = () => {
+    if (includeArchived) {
+      setIncludeActive(true)
+    } else {
+      setIncludeActive(false)
+    }
+    setIncludeArchived(!includeArchived)
+  }
+
+  // Close collapse when clicking anywhere in the window except the collapse items
+  window.addEventListener('click', () => setActiveKey(undefined), false)
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        whiteSpace: 'pre',
+        flexWrap: 'wrap',
+        maxWidth: '100vw',
+      }}
+    >
+      <Collapse
+        className={`projects-filter-collapse ${isDarkMode ? 'dark' : ''}`}
+        style={filterCollapseStyle}
+        activeKey={activeKey}
+      >
+        <CollapsePanel
+          style={{
+            border: 'none',
+          }}
+          key={0}
+          showArrow={false}
+          header={
+            <span
+              style={{ color: 'var(--text-secondary)' }}
+              onClick={e => {
+                setActiveKey(activeKey === 0 ? undefined : 0)
+                e.stopPropagation()
+              }}
+            >
+              <FilterOutlined /> <label>Filter</label>
+            </span>
+          }
+        >
+          {/* onClick: Do not close collapse when clicking its items*/}
+          <div onClick={e => e.stopPropagation()}>
+            {filterCheckboxItem('V1', includeV1, setIncludeV1)}
```

Comment:
**suggestion**: I think as a general rule, components should be rendered with JSX and not called as a function like this: https://kentcdodds.com/blog/dont-call-a-react-function-component

```suggestion
            <FilterCheckboxItem label='V1' checked={includeV1} onChange={setIncludeV1} />
```

To do this, the argument for `FilterCheckboxItem` needs to be a single object: https://github.com/jbx-protocol/juice-interface/pull/460/files#diff-3428316285ddea1141ee3a5a4378d2df3d2a0f6d92d8b72e7a6a47ab029e57d5R56

Reply:
Good call. Componentised it